language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/LSClientLogger.java

```diff
@@ -83,16 +84,20 @@ public void notifyUser(String operation, Throwable error) {
      * @param identifier text document
      * @param pos        pos
      */
-    public void logError(String message, Throwable error, TextDocumentIdentifier identifier, Position... pos) {
+    public void logError(LSOperation operation, String message, Throwable error, TextDocumentIdentifier identifier,
+                         Position... pos) {
         if (!this.isInitializedOnce) {
```

Comment:
I think we can return early if language client is `null` and get rid of repetitive null checks.
```suggestion
        if (!this.isInitializedOnce || this.languageClient == null) {
```